### ISSUE TITLE
Cant calculate some risk metrics.

### ISSUE THREAD

**Issue by acanacar** [quantopian/zipline#2414]
Dear Zipline Maintainers,

Before I tell you about my issue, let me describe my environment:

# Environment

* Operating System: (`ubuntu 16-04`)
* Python Version: `3.5.2`
* Python Bitness: `$ python -c 'import math, sys;print(int(math.log(sys.maxsize + 1, 2) + 1))'`
* How did you install Zipline: pip
* Python packages: 
alembic==1.0.5
bcolz==0.12.1
Bottleneck==1.2.1
certifi==2018.11.29
chardet==3.0.4
Click==7.0
contextlib2==0.5.5
cycler==0.10.0
cyordereddict==1.0.0
Cython==0.29.1
decorator==4.3.0
empyrical==0.5.0
et-xmlfile==1.0.1
h5py==2.9.0
idna==2.7
intervaltree==2.1.0
ipython-genutils==0.2.0
jdcal==1.4
johansen==0.0.4
jsonschema==2.6.0
jupyter-core==4.4.0
kiwisolver==1.0.1
Logbook==1.4.1
lru-dict==1.1.6
lxml==4.2.5
Mako==1.0.7
MarkupSafe==1.1.0
matplotlib==3.0.2
multipledispatch==0.6.0
nbformat==4.4.0
networkx==1.11
numexpr==2.6.8
numpy==1.15.4
openpyxl==2.5.12
pandas==0.18.1
pandas-datareader==0.5.0
patsy==0.5.1
plotly==3.5.0
pyparsing==2.3.0
python-dateutil==2.7.5
python-editor==1.0.3
pytz==2018.7
requests==2.20.1
requests-file==1.4.3
requests-ftp==0.3.1
retrying==1.3.3
scikit-learn==0.20.2
scipy==1.1.0
six==1.11.0
sklearn==0.0
sortedcontainers==2.1.0
SQLAlchemy==1.2.14
statsmodels==0.9.0
tables==3.4.4
toolz==0.9.0
trading-calendars==1.6.1
traitlets==4.3.2
urllib3==1.24.1
wrapt==1.10.11
xlrd==1.2.0
zipline==1.2.0

</details>

Now that you know a little about me, let me tell you about the issue I am
having:

# Description of Issue

At analyze layer after backtest i m getting all together metrics created on my own. Some of my metrics require; 
-count of trade with profit,
-count of trade with loss,
-sum of profit from trade with profit,
-sum of loss from trade with loss 

I have not been suffer from this issue since i have been using my own trailing stop function.And record every stages of trade (trade,stop changing,stoploss). But i have changed my strategy and starting to use StopLimitOrder. And rn i cant reach when my order crash to the stoploss or takeprofit.Thats why i cant calculate profit of these trades seperately.  

* What did you expect to happen?
* What happened instead?

Here is how you can reproduce this issue on your machine:

## What steps have you taken to resolve this already?
- Tried to benefit from performance orders metrics. But cant see cost base price. So i can calculate counts of them but net profits.
- Tried to getting it from transactions id but couldnt see cost base price again.
- Checked the positions metrics sinace it has cost base price and last sale price, but there is no stop loss or limit price. Additionally rn using minutely backtesting positions show me only the positions going on next day (yes i m using daily performance metrics.)
...

# Anything else?
In summary, if i get returned cost base price calling get_order(id) i would be problem-free.
...

Sincerely,
`acanacar`
